prometheus/CMakeLists.txt

```diff
@@ -22,12 +28,18 @@ add_custom_command(
   COMMAND protoc -I${CMAKE_SOURCE_DIR}/prometheus/proto
   ${CMAKE_SOURCE_DIR}/prometheus/proto/metrics.proto  --cpp_out=proto/)
 
-target_link_libraries(prometheus-client PUBLIC
+target_link_libraries(prometheus-client PRIVATE
```

Comment:
Adding the libraries to the public link interface also adds them to any other project that uses `prometheus-client`. That is, targets using `prometheus-client` like `integrations/gnu_libmicrohttpd/example` or `tests/without_bazel` not only receive `-lprometheus-client` but `-licu -lprotobuf` too. This is unnecessary since transient linkage is fine on both mac os and linux. That is, because `libprometheus-client` is already linked with `libicu` and `libprotobuf` we don't have to link them again. The linker picks them up for us.

With `PUBLIC` linkage downstream executables suddenly require `${ICU_LIBRARY_DIRS}` just to find `-licu` which is not actually needed...